src/Lecture1.hs

```diff
@@ -98,7 +99,16 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str
+    | end < 0   = ""
+    | start < 0 = take (end + 1) str
+    | otherwise = take (end - start + 1) (drop start str)
+
+{- Alternative one-line implementation:
+
+subString start end str = take (max (-1) end - max 0 start + 1) (drop start str)
```

Comment:
That's a nice one-liner! Although, it might be a bit hard to understand when reading this code later 😅